src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -268,52 +286,288 @@ internal BoundExpression ConvertPatternExpression(TypeSymbol inputType, CSharpSy
+            return new BoundRecursivePattern(
+                syntax: node, declaredType: boundDeclType, inputType: declType, deconstructMethodOpt: deconstructMethod,
+                deconstruction: patterns.ToImmutableAndFree(), propertiesOpt: propertiesOpt, variable: variableSymbol, variableAccess: variableAccess, hasErrors: hasErrors);
+        }
+
+        private BoundPattern BindPropertyPattern(PropertyPatternSyntax node, TypeSymbol operandType, bool hasErrors, DiagnosticBag diagnostics)
+        {
+            var typeSyntax = node.Type;
+            TypeSymbol declType = BindRecursivePatternType(typeSyntax, operandType, ref hasErrors, out var boundDeclType, diagnostics);
+            ImmutableArray<(Symbol property, BoundPattern pattern)> propertiesOpt = BindPropertySubpattern(node.PropertySubpattern, declType, diagnostics, ref hasErrors);
+            BindPatternDesignation(node, node.Designation, declType, typeSyntax, diagnostics, ref hasErrors, out var variableSymbol, out var variableAccess);
+            return new BoundRecursivePattern(
+                syntax: node, declaredType: boundDeclType, inputType: declType, deconstructMethodOpt: null,
+                deconstruction: default, propertiesOpt: propertiesOpt, variable: variableSymbol, variableAccess: variableAccess, hasErrors: hasErrors);
+        }
+
+        ImmutableArray<(Symbol property, BoundPattern pattern)> BindPropertySubpattern(
+            PropertySubpatternSyntax node,
+            TypeSymbol inputType,
+            DiagnosticBag diagnostics,
+            ref bool hasErrors)
+        {
+            var builder = ArrayBuilder<(Symbol property, BoundPattern pattern)>.GetInstance();
+            foreach (var p in node.SubPatterns)
+            {
+                var name = p.NameColon?.Name;
+                var pattern = p.Pattern;
+                Symbol property = null;
+                TypeSymbol propertyType;
+                if (name == null)
+                {
+                    if (!hasErrors)
+                    {
+                        diagnostics.Add(ErrorCode.ERR_PropertyPatternNameMissing, pattern.Location, pattern);
+                    }
+
+                    propertyType = CreateErrorType();
+                    hasErrors = true;
+                }
+                else
+                {
+                    property = LookupPropertyForPattern(inputType, name, out propertyType, ref hasErrors, diagnostics);
+                }
+
+                var boundPattern = BindPattern(pattern, propertyType, hasErrors, diagnostics);
+                builder.Add((property, boundPattern));
+            }
+
+            return builder.ToImmutableAndFree();
+        }
+
+        private Symbol LookupPropertyForPattern(TypeSymbol inputType, IdentifierNameSyntax name, out TypeSymbol propertyType, ref bool hasErrors, DiagnosticBag diagnostics)
```

Comment:
>Property [](start = 29, length = 8)

The use of the "Property" word in the name of the function, name of ```propertyType``` parameter and in local names in consumers is misleading. Consider replacing with "Member" or "PropertyOrField"

Reply:
Resolved in [https://github.com/dotnet/roslyn/pull/23357/commits/758fd76932a6d9bcb640504c8fdef57fb5adecec](https://github.com/dotnet/roslyn/pull/23357/commits/758fd76932a6d9bcb640504c8fdef57fb5adecec) of [https://github.com/dotnet/roslyn/pull/23357](https://github.com/dotnet/roslyn/pull/23357)

---
In reply to: [156521610](https://github.com/dotnet/roslyn/pull/23209#discussion_r156521610) [](ancestors = 156521610)